libyul/backends/evm/StackLayoutGenerator.cpp

```diff
@@ -549,6 +620,55 @@ Stack StackLayoutGenerator::combineStack(Stack const& _stack1, Stack const& _sta
 	return commonPrefix + bestCandidate;
 }
 
+vector<StackLayoutGenerator::StackTooDeep> StackLayoutGenerator::reportStackTooDeep(CFG::BasicBlock const& _entry) const
+{
+	vector<StackTooDeep> stackTooDeepErrors;
+	util::BreadthFirstSearch<CFG::BasicBlock const*> breadthFirstSearch{{&_entry}};
+	breadthFirstSearch.run([&](CFG::BasicBlock const* _block, auto _addChild) {
+		Stack stack;
+		stack = m_layout.blockInfos.at(_block).entryLayout;
+
+		for (auto const& operation: _block->operations)
+		{
+			Stack& operationEntry = m_layout.operationEntryLayout.at(&operation);
+
+			stackTooDeepErrors += findStackTooDeep(stack, operationEntry);
+			stack = operationEntry;
+			for (size_t i = 0; i < operation.input.size(); i++)
+				stack.pop_back();
+			stack += operation.output;
```

Comment:
I originally wanted to say that `stack` gets overwritten in every cycle so this looks pointless but then I noticed that it still gets passed to `findStackTooDeep()` in the next loop cycle. Maybe it would be a good idea to rename it to make its purpose clearer? E.g. `previousOperationExitLayout`?

Reply:
Renamed it to ``currentStack`` now, maybe that's clearer?

Reply:
Well, part of the problem for me was that I was treating it as the current stack but at some points in the code (especially just before this) it's easier to actually think about it as the previous stack.

At least it's better than just `stack` :)